src/components/document/document-group.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useRef, useState } from "react";
+import { observer } from "mobx-react-lite";
+import { DocumentModelType, getDocumentContext } from "../../models/document/document";
+import { SimpleDocumentItem } from "../thumbnail/simple-document-item";
+import { DocumentContextReact } from "./document-context";
+import { IDocumentMetadata } from "../../../functions/src/shared";
+import { DocumentGroup } from "../../models/stores/document-group";
+
+import ScrollArrowIcon from "../../assets/workspace-instance-scroll.svg";
+
+import "./document-group.scss";
+
+interface IProps {
+  documentGroup: DocumentGroup;
+  secondarySort: string;
+  onSelectDocument: (document: DocumentModelType | IDocumentMetadata) => void;
+}
+
+export const DocumentGroupComponent = observer(function DocumentGroupComponent(props: IProps) {
+  const { documentGroup, secondarySort, onSelectDocument } = props;
+  const docBoxWidth = 16;
+  const docBoxGap = 10;
+  const scrollUnit = docBoxWidth + docBoxGap;
+  const docCount = documentGroup.documents.length || 0;
+  const isUnsorted = secondarySort === "None";
+  const docListContainerRef = useRef<HTMLDivElement>(null);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [visibleCount, setVisibleCount] = useState(0);
+  const [leftArrowDisabled, setLeftArrowDisabled] = useState(true);
+  const [rightArrowDisabled, setRightArrowDisabled] = useState(false);
+
+  // Each document in the group is represented by a square box. The group of document boxes is displayed in
+  // a single row. If there are more boxes than can fit within the row's width, scroll buttons are added
+  // to either side of the list so the user can scroll through it.
+  const handleScroll = (direction: "left" | "right") => {
+    if (docListContainerRef.current) {
+      const scrollAmount = visibleCount * scrollUnit;
+      docListContainerRef.current.scrollBy({
+        left: direction === "left" ? -scrollAmount : scrollAmount,
+        behavior: "smooth"
+      });
+    }
+  };
+
+  // Set up a resize observer for responding to changes to the document list container's width.
+  useEffect(() => {
+    const docListContainer = docListContainerRef.current;
+
+    const updateWidth = () => {
+      if (docListContainer) {
+        setContainerWidth(docListContainer.offsetWidth);
+      }
+    };
+
+    updateWidth();
+
+    const resizeObserver = new ResizeObserver(() => {
+      updateWidth();
+    });
+
+    if (docListContainer) {
+      resizeObserver.observe(docListContainer);
+    }
+
+    return () => {
+      if (docListContainer) {
+        resizeObserver.unobserve(docListContainer);
+      }
+    };
+  }, []);
```

Comment:
This `useEffect` should only run once to set up the resize observer when the component mounts. So it has an empty dependency array. Also, I believe you don't usually add refs to `useEffect` dependency arrays since changes to refs don't trigger re-renders.

Reply:
Ah right, my mistake.